### PR TITLE
Doh's solution #1

### PR DESCRIPTION
**Pull request by djdoh88**
Gonna work a bit more later lol

### CODE REVIEW

File: doh_solution.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
+# coding: utf-8
+
+# In[16]:
+
+import re
```
Comment (djdoh88) [inline doh_solution.py]: I chose to use regex, was that a good idea?

File: doh_solution.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+# In[17]:
+
+def longest_substring(s):
+
```
Comment (williamsmj) [inline doh_solution.py]: Please include a docstring

**Comment by mikepqr** [comment]
Hey @djdoh88/@dohbrandwatch, couple of problems:

It fails the test harness because it returns the longest substring _and_ the length of the longest substring as a tuple. It should return only the length!

It also fails a couple of manual tests:

```
In [11]: doh_solution.longest_substring('a')   # should be 1!
Out[11]: []

In [13]: doh_solution.longest_substring('abcdabc')  # should be 4!
Out[13]: [('cdabc', 5), ('bcdab', 5), ('abcda', 5)]
```

**Comment by mikepqr** [comment]
I guess you've abandoned this, right? For future reference, there's no need to do that (and indeed it's semantically the wrong thing to do!)

Anyway, closing :smile:

**Comment by djdoh88** [comment]
I think I meant to abandon my old one yes! Thanks!!

**Pull request closed without merging**